### DISCUSSION TITLE
compare struct and interface obj

### DISCUSSION THREAD

**Discussion by getfile** [vlang/v#17260] [Questions and Answers]
struct Data {
     v f32
}

a:=Data{}
b:=Data{}
assert a==b     //is true

it compare its value,  now i want to compare it's addtress
&u8(&a)==&u8(&b) is ugly.  and not work for interface object
how can i do that? 
may be add a operator "===" ?

**Comment by MatejMagat305** [comment]
well I don't know it can be done with `unsafe { voidptr(&a) == voidptr(&b) }`, it is not the most beautiful code what I have seen, but it is useable ...

**Reply by getfile** [reply]
yes , it work for struct , but not work for interface object. for example:

interface IData { v f32 }
mut datas := []IData{}
a:=Data{}
b:=Data{}
datas << a
datas << a
datas << b
assert datas[0]==datas[1] //true
assert datas[0]==datas[2] //true

[0] and [1] is same object, not problem
[0] adn [2] is not same object , i want result of compare is false.

so how can i solve this problem? voidptr(&) not work for this situation

**Reply by JalonSolov** [reply]
In V, an interface is not an object.  It's a contract that structs may follow to allow them to be passed to the same routines, even if there are some differences.

**Reply by getfile** [reply]
yes , i check the code in c version, interface obj is transform into struct obj, and enclose original struct data.
so from interface obj, v can get the struct data addtress, but its private from programmer.
so v can give some kind of solution for human?
